File: app/src/main/java/com/example/punchpad2/NoteEntity.java
package com.example.punchpad2;

import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.ColumnInfo;
import androidx.room.Ignore;


@Entity(tableName = "notes")
public class NoteEntity {

    @PrimaryKey(autoGenerate = true)
    public int id;

    @Ignore
    public boolean expanded = false;

    public String content; // full text content, image/audio tags inline

    public long createdAt;
    public long lastEdited;

    public boolean isHidden;
    public boolean isLarge;

    @ColumnInfo(name = "folder_id")
    public long folderId;

    public boolean isTrashed;
    public boolean favorited;

    // future fields can be added here with migration
}

File: app/src/main/java/com/stratonotes/NoteEntity.kt
package com.stratonotes

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.ColumnInfo
import androidx.room.Ignore

data class MediaItem(
    val type: String, // "image" or "audio"
    val uri: String   // URI string pointing to the media file
)

@Entity(tableName = "notes")
data class NoteEntity(

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long = 0L,

    @ColumnInfo(name = "folderId")
    val folderId: Long,

    @ColumnInfo(name = "content")
    var content: String,

    @ColumnInfo(name = "createdAt")
    val createdAt: Long = System.currentTimeMillis(),

    @ColumnInfo(name = "lastEdited")
    var lastEdited: Long = System.currentTimeMillis(),

    @ColumnInfo(name = "isFavorite")
    var isFavorite: Boolean = false,

    @ColumnInfo(name = "isHiddenFromMain")
    val isHiddenFromMain: Boolean = false,

    @ColumnInfo(name = "isLarge")
    val isLarge: Boolean = false,

    @ColumnInfo(name = "isTrashed")
    var isTrashed: Boolean = false

) {
    @Ignore
    var mediaItems: MutableList<MediaItem> = mutableListOf() // UI-only field

    override fun equals(other: Any?): Boolean {
        return other is NoteEntity && other.id == this.id
    }

    override fun hashCode(): Int = id.hashCode()
}
